### ISSUE TITLE
Spanish translation (buena suerte)

### ISSUE THREAD

**Issue by goldbergyoni** [goldbergyoni/nodebestpractices#95]
https://github.com/i0natan/nodebestpractices/blob/spanish-translation/README.md

@eduardomontalvo - Spanish translation can be officially kicked-off. You're the lead! I'll be at your disposal should you need anything. A branch was created (link below), you should have permissions to it, everyone else must PR before changing anything there.

I would ask/recommend only 2 important guidelines:

Keep the same content & formatting - the focus is on translation, should anyone want to modify the content or the graphics - let's PR first in English and them translate to Spanish
Duplicate pages - the content should be translated over a page duplication. Readme.MD became Readme.spanish.MD, all other files should be duplicated similarly. So the number of English & Spanish pages should be the same. Will create a bold link from the English home page to the Spanish home page (with bold credit of course...)

https://github.com/i0natan/nodebestpractices/blob/spanish-translation/README.md

**Comment by i0natan** [comment]
@eduardomontalvo no rush, just checking whether everything is clear and if I assist in anyway?

**Comment by i0natan** [comment]
@eduardomontalvo 

Greetings.

I met a guy in another project who's doing Spanish translation, he's just not technical/Node-developer. He's willing to translate all the sections (inner pages) so you're left to: (1) translate only the home page - which I noted you already did most of it (2) verify his work. 

Are you available these days for this?

**Comment by desko27** [comment]
Hi @i0natan,

I was just checking again this project since its advices are being useful to me for my current project. I just noticed it's still not fully translated to my mother language, but I see that you already have plans for it.

Let me know if I may help in any way.

**Comment by i0natan** [comment]
@desko27 Welcome aboard :]

Is your mother tongue language Spanish?

If yes, we will really appreciate some help - Already have the home page 70% translated and a guy who can translate all inner page. Just need someone who can coordinate this work.

**Comment by desko27** [comment]
@i0natan Thanks!

Yes it is, I'm from Barcelona, Spain.

Wow 70% translated at the home page, that's good news! So what do you mean by coordinating this work? Are you talking about setting some conventions/standards? By the other hand, I'll be happy to collaborate by reviewing the translations if that's helpful. Would that be ok?

**Comment by i0natan** [comment]
Hey @desko27, Barcelona is good! :)

Coordinate = you lead the Spanish translation the way you like. You have 2 resources that you might or not use: (1) this PR translated part of the home page, it's around 50-80% completed (2) there is a guy who's willing to translate all inner-pages in few days, but he's not technical so probably have to go through this and ensure that the translation is appropriate

What do you think?

The Spanish PR: https://github.com/i0natan/nodebestpractices/pull/16/files

**Comment by desko27** [comment]
@i0natan I see! I'll keep an eye on that PR then. Thanks!

**Comment by i0natan** [comment]
@desko27 So you wanna take the Spanish translation and handle it? should I connect you with the guy who can translate all the inner pages (then you just need to go over it and approve)?

Your name of course will appear on top of the Spanish page + on the English repo main page

**Comment by desko27** [comment]
@i0natan Yeah, that would be fantastic, my skype/discord ID is desko27.

**Comment by i0natan** [comment]
@desko27 Great! The guy will be available starting from Tuesday next week, I'll make sure to schedule a mutual session. He can carry the heavy lifting, make the first translation, and you as a developer can ensure that the technical lingo isn't lost in translation

**Comment by i0natan** [comment]
@desko27 Cool, pinging on skype tomorrow

**Comment by i0natan** [comment]
@desko27 Added on Skype

**Comment by i0natan** [comment]
@desko27 Added to the Spanish branch the partial homepage translation that was previously done + a Spanish file for each inner page (e.g. sections/errorhandling/asyncerrorhandling.spanish.md) so you can paste the translation. Can I help in any other way?

https://github.com/i0natan/nodebestpractices/tree/spanish-translation

**Comment by desko27** [comment]
@i0natan Hi, I took a look today and saw that the guy who is translating the inner pages translated a few of them. I have some personal issues to attend to this week. Next week on monday/tuesday I'll be updating this with the existing translations, and I may translate some of the remaining translations in the homepage.

Just a detail, [this spanish inner page](https://github.com/i0natan/nodebestpractices/blob/spanish-translation/sections/projectstructre/breakintcomponents.spanish.md) was not copied from english but chinese.

**Comment by i0natan** [comment]
@desko27 Cool, you're the captain, do it on your convenient time and take the ship to Costa Brava. Just if possible, Seth has asked few questions in the Excel, can you answer him as he wishes to accomplish his work?

Sorry about the inner page issue, I guess it's only 1 and the rest are good

**Comment by i0natan** [comment]
@desko27 Just a minor question - do we need anything else from Seth? did he did his job?

**Comment by desko27** [comment]
Hey @i0natan !!

I only see around 25% of translations done in our shared document. Should I wait for Seth to complete his work or should I review & PR the existing translations?

By the other hand, last changes were made by Seth 20 days ago. We maybe should ask him if everything is ok?

**Comment by i0natan** [comment]
@desko27 Let's approach him in Skype and ask about this... I was sure that we was done with the work

**Comment by i0natan** [comment]
@desko27 I managed to catch Seth, it wasn't easy, he said that we disappeared... :]

Practically, he's willing now to finalize but have one concern: "As a note, in the excel it shows 46 articles or so but the links are 30 something, do you have the other links so we can translate those articles too? "

Let me know what should I answer?

**Comment by desko27** [comment]
@i0natan that's cool!

About the links, I'll check it when I get home. Some articles had no one-paragraph explanation, so it's probably that.

**Comment by i0natan** [comment]
@desko27 Thanks for the vast answer, let's make him finalize this quick.

Maybe you can add more articles from the security branch which will get PRed soon (6.1-6.22):
https://github.com/i0natan/nodebestpractices/tree/security-best-practices-section#6-security-best-practices

**Comment by i0natan** [comment]
@desko27 I had hard time to get his attention, any chance we just push the links pasting so he can finalize his work?

**Comment by desko27** [comment]
@i0natan this is the report:

- He translated linked articles from 1 to 11 with some red marks which I'll take care of without any extra action required from him, so we can consider them as finished work from him.
- **There're untranslated linked articles from 12 to 34** which he can freely translate whenever he likes (it's been like this from the day I pasted them).
- I will be pasting new links from [that branch](https://github.com/i0natan/nodebestpractices/tree/security-best-practices-section#6-security-best-practices) soon.

So I can paste new links soon, but there's at least 2/3 of pending work already which doesn't depend on us. In other words, we're not blocking him in any way.

Also, as a reminder, the current links are the only ones that had a one-paragraph explanation, that's why there's only 34.

**Comment by i0natan** [comment]
You're absolutely right in every word. He just wants to perform all in one shot, so he gets all the links and get down on it. He doesn't want to perform few iterations, so the "blocking" artifacts are the rest of the links.

**Comment by desko27** [comment]
@i0natan Ok, I understand. I'll be home in 2-3h from now, so I'll ensure to provide the new links from the security branch by then. With that done he can perform the one shot 👍

**Comment by desko27** [comment]
@i0natan All new links from the security branch are now inserted into the document. There's a total of 51 links. I'll be watching for any question he may have.

**Comment by i0natan** [comment]
@desko27 Great, the ball is now in Seth's court. 

Do we want Seth to handle the main page as well or do you wanna take this?

p.s. Good luck Spain in the 1/8 finals :]

**Comment by desko27** [comment]
@i0natan I can take on the main page, I'll start this week.

Linked articles from the 12th one remain untranslated.

p.s. I'm sorry I'm not a big fan of soccer haha.

**Comment by desko27** [comment]
@i0natan Hey!

- Articles are translated from 1 to 11 (like before) AND from 35 to 51.
- Articles are NOT translated from 12 to 34.

I checked a few times and all links are working fine.

**Comment by i0natan** [comment]
@desko27 Great catch. Asking for fixes. Can you remind me the link?

**Comment by desko27** [comment]
@i0natan I sent it to you via skype since everyone with the link can edit the document.

**Comment by i0natan** [comment]
@desko27 Yeah, forgot that... thanks

**Comment by i0natan** [comment]
@desko27 You were right, they forgot to translate this part... It's now all transalated. I verified myself

If you can easily define how to copy from the Excel to a GH file - I've someone who's doing data entry and can do this for us. Do you want to instruct her or do yourself?

**Comment by desko27** [comment]
@i0natan Nice! I would like to do it myself, since I wanted to review each translation as well so we make sure it sounds as natural as possible. Can I open a PR as soon as I get it? It shall be done between this and next week (I would like to do it sooner but I have some conferences this weekend I must attend to). I love to strive to perfection with this kind of stuff so I promise it's worth it 👍

**Comment by i0natan** [comment]
Sounds like that perfectionism I couldn't resist 👍 👍 👍 

Do you need anything else from my part? you can open a PR whenever you like

**Comment by js-kyle** [comment]
Hey there @desko27!

I'm checking in on how progress is going with the Spanish translation. I'm taking a look at the current pull requests open for this

1. https://github.com/i0natan/nodebestpractices/pull/16 Add spanish translation skeleton
2. https://github.com/i0natan/nodebestpractices/pull/264 translate writing-guidelines to spanish
3. https://github.com/i0natan/nodebestpractices/pull/282 Update READMEspanish.md

How can I help progress this? Let me know if there is anything I can do. The third PR looks like a lot of work has been done towards this!

**Comment by gianpaj** [comment]
what's the current status? is there a specific task/PR that's ready to be worked on?
I can't understand which one is most important and which will allow the others to unblock

**Comment by js-kyle** [comment]
Hi @gianpaj - it really is a question for @desko27, as he is the lead of the Spanish translation work. Although he has not checked in here for some time, so we may need to reassign, unless he is still available

@desko27 would you be able to confirm if you are still able to dedicate some time here, or if you are too busy?

**Comment by palafoxernesto** [comment]
Hey guys, im ready to dedicate some time working on translation. 
@desko27 Are you still leading this effort? 
If yes, Maybe @gianpaj (if you're still interested) and I can help you.
Thanks

**Comment by desko27** [comment]
Hi guys, It's been a while so my apologies for that!... I'm immediately opening a PR with the current progress which is half of translations I reviewed from Seth, so you can go on from there.

I've been through a lot these months and I lack the time to finish this ATM, may you @gianpaj and @palafoxernesto continue with the task? If that's the case I can provide you the link to the rest of unreviewed translations via Skype (my skype id is desko27).

**Comment by gianpaj** [comment]
Thanks for the PR @desko27 

I had some time back then. I should have some time next month, in April

**Comment by desko27** [comment]
@gianpaj I added a task list 📝 about the current progress on #363's description. Despite being short in time, I'll try to continue and review the remaining translations by following down the task list and contributing to the mentioned PR (whatever my time allows me to finish).

Feel free to comment whenever you have some time to contribute and I'll try to take a moment so we can organize the collaboration. Thanks!!

**Comment by i0natan** [comment]
@desko27 sounds great!

**Comment by js-kyle** [comment]
@desko27 is this PR still of any use to you? Or should we close it for clarity? https://github.com/i0natan/nodebestpractices/pull/16/files

**Comment by desko27** [comment]
@js-kyle #16 has useful work done in the `README.md` file, but the author is not active on it anymore I fear. I see the PR has been automatically closed due to inactivity. I could review the contents and push them as a new PR while mentioning their original author.

**Comment by desko27** [comment]
Hey guys, I'm finishing off #363 on August, I can finally have some schedule for it... stay tuned.

**Comment by js-kyle** [comment]
Sounds great!

**Comment by desko27** [comment]
@goldbergyoni @js-kyle, about #363, I already completed all "one paragraph explainers" for sections from 1 to 5 👌

I would like to start section 6 (Security Best Practices), but since it was included to `master` after spanish translation started, files are not present on the branch I'm working on, which is [`spanish-translation`](https://github.com/goldbergyoni/nodebestpractices/tree/spanish-translation). Can you handle the merging from master to spanish-translation branch, and then create the `*.spanish.md` files for section 6 one-paragraph-explainer files? Or should I do that myself in the #363 as well?

**Comment by js-kyle** [comment]
Hey @desko27 Awesome! Regarding getting started on Section 6, I think it might be best to merge the current branch to master, then create a fresh start for part 6? Whichever option makes it easier for you. If you have a preference, let us know here and we can arrange that

**Comment by desko27** [comment]
@js-kyle that's cool! so this is the roadmap, in this specific order:
1. #363 can be merged into [spanish branch](https://github.com/goldbergyoni/nodebestpractices/tree/spanish-translation).
2. I need `master` to be merged into [spanish branch](https://github.com/goldbergyoni/nodebestpractices/tree/spanish-translation) because it's outdated and section 6 is missing.
3. I'll open a new PR for section 6 translations.

**Comment by js-kyle** [comment]
Hey @desko27, I think this is ready to continue work on a new branch now. Please let me know if the merge with master is what you were expecting, otherwise, I can fix it!

**Comment by desko27** [comment]
@js-kyle I see, that's perfect 😄

I'm busy for the next two weeks. So after that, I'm working on a new PR ✍️ for section 6 🔒 Security Best Practices.

**Comment by goldbergyoni** [comment]
@js-kyle @desko27 If the 5 sections are merged with master - I would release that, put a Spanish flag and include in news. What do you think?

**Comment by Zybax** [comment]
Hey so.. i was looking for some projects to start contributing and this one looks beginner friendly,
is there something i need to know before contributing?

**Comment by js-kyle** [comment]
@Zybax that's awesome, welcome! Were you interested in helping with the Spanish translation?

**Comment by Zybax** [comment]
@js-kyle yeah i have seen some issues already, also im getting into node and this looks  like a great source on info i'll make sure to read it all

**Comment by js-kyle** [comment]
The current working branch for spanish is https://github.com/goldbergyoni/nodebestpractices/tree/spanish-translation

So you can create a fork of this repository for you to work on, then once you have some progress, we will be happy to review it.

@desko27 is leading the Spanish translation, and has currently worked through sections 1 - 5, and will work on section 6 soon I believe. Perhaps you could work on some items for section 6, and he could review it once ready

**Comment by Anonyma** [comment]
Hi! Any pending translations? 
Is anyone working on these already? (security section) https://github.com/goldbergyoni/nodebestpractices/commit/a4d3d0585e8d4b1b9db3237537a3f6f47f3a856e

**Comment by desko27** [comment]
Hi @Zybax @Anonyma! I'm sorry, I've been a little bit busy. I'll try to expose the current status of the translation project so you can decide what's the best way for you to contribute:

## One paragraph explainers
There's no much left to do with these ones. Anything is improvable by the community though.
- ✅ Sections 1 to 5 are translated and merged.
- 🔃 Section 6 is almost finished and I'll open a PR, you could help by reviewing it if you wish.

## Main readme file [[link]](https://github.com/goldbergyoni/nodebestpractices/blob/spanish-translation/README.spanish.md)
There's still work to be done and help is appreciated.
- 🆘 Sections 1 and 2 are translated, but I think they're improvable like you did in #577.
- 🆘 Sections 3 to 6 are completely untranslated, they need to be translated.

Also, the english contents present in [spanish main file](https://github.com/goldbergyoni/nodebestpractices/blob/spanish-translation/README.spanish.md) are a bit outdated, so translations should be done by looking at [the updated english main file](https://github.com/goldbergyoni/nodebestpractices/blob/master/README.md) instead. We could also "update" the spanish main file with the latest english text before going any further.

**Comment by goldbergyoni** [comment]
@desko27 Awesome summary. Whenever you feed the Spanish translation is good enough to be published - let us know. I don't think it should be perfect, we're a community-based project so even if it's 85% ready - let spread the word

@Zybax @Anonyma Welcome abroad, we will be glad to have you hear on improving the content! 🎉

**Comment by Whitefox98** [comment]
I already translated until point 7.2 . I'm spanish native

https://github.com/goldbergyoni/nodebestpractices/pull/607

**Comment by desko27** [comment]
That's awesome @Whitefox98 :clap: :clap: I've been struggling to get some time for this and I was expecting to use holidays. I'll help with the review ASAP! Thank you!

**Comment by Whitefox98** [comment]
No problem, Thanks to you to help the comunity! @desko27

**Comment by jormaechea** [comment]
Hi @goldbergyoni, are you still working on this? I saw that the home is 100% translated, but not the inner pages.

I'm spanish native and I could give some help.
Let me know.

**Comment by desko27** [comment]
@jormaechea I've been translating the project myself in the past but I have no time to keep doing it right now, so any contribution is appreciated. I'm just helping with reviews, so feel free to identify what's missing, send a PR and I'll take a look in order for it to be merged ASAP 👍🏻 Thank you so much.

**Comment by jormaechea** [comment]
Great @desko27, I’ll try to spend some time on this. Can we reopen the issue?

**Comment by kevynb** [comment]
Hello @jormaechea, welcome to node best practices and thanks for offering to contribute 😀
No need to keep this issue open. Open a pull request with your changes targeting the branch that contains the Spanish translation (I don’t remember if it’s master and I’m on my phone right now) and tag @desko27. As soon as he’s okay with your changes, we’ll approve and merge it! 🌠
